INTERNSHIP/env/Lib/site-packages/pip/_internal/network/download.py

```
@@ -0,0 +1,186 @@
+        self._session = session
+        self._progress_bar = progress_bar
+
+    def __call__(
+        self, links: Iterable[Link], location: str
+    ) -> Iterable[Tuple[Link, Tuple[str, str]]]:
+        """Download the files given by links into location."""
+        for link in links:
+            try:
+                resp = _http_get_download(self._session, link)
+            except NetworkConnectionError as e:
+                assert e.response is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)